src/Drupal/Commands/core/LocaleCommands.php

```diff
@@ -144,76 +145,121 @@ public function update($options = ['langcodes' => self::REQ])
      *   Export the Dutch non-customized translations.
      * @usage drush locale:export --langcode=nl --types=customized,not-customized > nl.po
      *   Export the Dutch customized and not customized translations.
-     * @usage drush locale:export > drupal.pot
+     * @usage drush locale:export --template > drupal.pot
      *   Export the template file to translate.
      * @aliases locale-export
      * @see \Drupal\locale\Form\ExportForm::submitForm
      *
      * @throws \Exception
      */
-    public function export($options = ['langcode' => self::OPT, 'types' => self::OPT])
+    public function export($options = ['template' => false, 'langcode' => self::OPT, 'types' => self::OPT])
     {
+        $language = null;
+        $poreader_options = null;
+        $template = (bool)$options['template'];
+
         // If template is required, language code is not given.
-        if ($options['langcode'] != LanguageInterface::LANGCODE_SYSTEM) {
-            $language = \Drupal::languageManager()->getLanguage($options['langcode']);
-
-            if ($language == null) {
-                $this->logger()->warning(dt('Language code @langcode is not configured.', [
-                    '@langcode' => $options['langcode'],
-                ]));
-                return;
-            }
-        } else {
-            $language = null;
+        if ($options['langcode'] != null) {
+            $language = $this->getLanguage($options['langcode']);
+        } elseif ($template === false) {
```

Comment:
`if ($options['langcode'] != null) {`
Move this logic into ::getLanguage() (or delete it) and base the error message on the value of $language.